writeup/src/body.lhs

```diff
@@ -4,149 +4,160 @@
 % Bug because of double lines?
 % https://blog.codecentric.de/en/2014/02/curly-braces/
 
-  Software Version Control Systems are an essential tool in the 
-belt of today's software engineer. At their core is a
-differencing algorithm that computes patches between two versions of a file, 
-the most well-known being the UNIX \texttt{diff}~\cite{McIlroy1976}.
-It is a line-based tool, i.e., it looks at changes
-on the level of the \emph{lines} of a file, hence, it fails
-to identify more fine grained changes in software source code. 
-
-  In this paper we go over our representation of changes between
-arbitrary trees and an efficient algorithm to compute them. These
-can be particularly applied to abstract syntax trees and offer a
-different approach to look at file differencing. We have implemented
-our work in Haskell. The type-safety and generic programming capabilities
-of functional languages make it the optimal tool for approaching
-this task.
-
-  Abstractly, the well-typed differencing problem for some type |a|
-consists in finding a type constructor, call it |Patch a|, together with
-functions |diff| and |apply|. The |diff| function computes
-the differences between two values of type |a|, and |apply|
-attempts to transform one value according to the information stored in the |Patch|. 
-\emph{Well-typed}, here, refers to the fact that a |Patch| is guaranteed, by
-construction, to generate well-typed elements once it is applied. This
-is not the case for the approaches~\cite{Asenov2017,Falleri2014} using
-\texttt{xml} or \texttt{json} to represent their abstract syntax
-trees, where the result of an arbitrary patch could be an ill formed tree.
-
+Software Version Control Systems are an essential tool in modern
+software development. Platforms such as Github and Bitbucket are only
+possible due to the widespread adoption of tools such as git,
+mercurial and darcs, that enable multiple developers to collaborate
+effectively.  At the heart of all these tools is the UNIX \texttt{diff}
+utility~\cite{McIlroy1976}, that computes a patch between two versions
+of a file.  The \texttt{diff} utility compares files on a line-by-line
+basis and attempts to share lines between its source and destination
+whenever possible. While this works well in practice, this fixed
+granularity of change cannot describe common refactorings---such as
+variable renaming---effectively. Representing all patches on the basis
```

Comment:
Neither can we. If it works well in practice, then why are we doing it differently?